Source/WebKit/Shared/Cocoa/SandboxExtensionCocoa.mm

```diff
@@ -333,29 +336,21 @@ String resolvePathForSandboxExtension(StringView path)
         return std::nullopt;
     }
     
-#if HAVE(SANDBOX_STATE_FLAGS)
-    // When launchd is blocked in the sandbox, we need to manually enable bootstrapping of new XPC connectons.
-    // This is done by unblocking launchd, since launchd access is required when creating Mach connections.
-    // Unblocking launchd is done by enabling a sandbox state variable.
-    if (machBootstrapOptions == MachBootstrapOptions::EnableMachBootstrap) {
-        if (auditToken) {
-            if (!sandbox_enable_state_flag(ENABLE_MACH_BOOTSTRAP, *auditToken))
-                RELEASE_LOG_FAULT(Sandbox, "Could not enable Mach bootstrap, errno = %d.", errno);
-        } else
-            RELEASE_LOG_FAULT(Sandbox, "Could not enable Mach bootstrap, no audit token provided.");
-    }
-#endif
-
     return WTFMove(handle);
 }
 
 auto SandboxExtension::createHandlesForMachLookup(Span<const ASCIILiteral> services, std::optional<audit_token_t> auditToken, MachBootstrapOptions machBootstrapOptions, OptionSet<Flags> flags) -> Vector<Handle>
 {
-    return createHandlesForResources(services, [auditToken, machBootstrapOptions, flags] (ASCIILiteral service) -> std::optional<Handle> {
-        auto handle = createHandleForMachLookup(service, auditToken, machBootstrapOptions, flags);
+    auto handles = createHandlesForResources(services, [auditToken, flags] (ASCIILiteral service) -> std::optional<Handle> {
+        auto handle = createHandleForMachLookup(service, auditToken, flags);
         ASSERT(handle);
         return handle;
     });
+
+    if (machBootstrapOptions == MachBootstrapOptions::EnableMachBootstrap)
+        handles.insert(0, createHandleForMachBootstrapExtension());
```

Comment:
Is there any particular significance to order in this vector?

If it's important for the mach bootstrap handle to be first, can we put it in the vector before we call createHandlesForResources rather than after?

Reply:
> Is there any particular significance to order in this vector?
> 
> If it's important for the mach bootstrap handle to be first, can we put it in the vector before we call createHandlesForResources rather than after?

That is a good point. The order is not important in this vector. I have changed this to append instead.